### ISSUE TITLE
SnowflakeDbDataReader.HasRows() always returns true

### ISSUE THREAD

**Issue by DR486** [snowflakedb/snowflake-connector-net#413]
### Issue description
SnowflakeDbDataReader.HasRows() should return true when there are rows, and false when there are no rows.  [Currently it's only returning true.  ](https://github.com/snowflakedb/snowflake-connector-net/blob/80b595164ce89c6678b3d16ec5e99610edb65caa/Snowflake.Data/Client/SnowflakeDbDataReader.cs#L71)

The comment for that function states that one row will always be returned. When using `while(reader.Read())` to read the results of a select query when no records are found, the while loop is skipped.

### Example code
```cs
//Always returns true, see https://github.com/snowflakedb/snowflake-connector-net/blob/80b595164ce89c6678b3d16ec5e99610edb65caa/Snowflake.Data/Client/SnowflakeDbDataReader.cs#L71
var exists = reader.HasRows(); 

while(reader.Read()){
//This (obviously) never executes if the query executed returns no results....  
//but that means that reader.HasRows() should indeed return false.
   var test = reader.GetValue(0); 
}


```

### Error log
```
N/A
```
There is an example in READMD.md file showing you how to enable logging.

### Configuration
*Driver version:* 2.0.8

*Dotnet framework and version:* .NET Core 3.1 & .NET 6

*Server version:* N/A
You may get the server version by running a query:
```
SELECT CURRENT_VERSION();
```

*Client OS:* Windows 10

**Comment by Christian-Oleson** [comment]
Here is where this is hard coded:
https://github.com/snowflakedb/snowflake-connector-net/blob/master/Snowflake.Data/Client/SnowflakeDbDataReader.cs#L77

**Comment by JCurto** [comment]
This is still being hard coded to true for all SELECT statement types (and others)?:
https://github.com/snowflakedb/snowflake-connector-net/blob/master/Snowflake.Data/Core/ResultSetUtil.cs#L72

**Comment by sfc-gh-jtang** [comment]
@sfc-gh-igarish can you double check to see if it's fixed already?

**Comment by sfc-gh-dszmolka** [comment]
hi and thank you for submitting this issue, also apologies for the long period without response! at least can post some good news now; this seems to be fixed with #665 and the fix is merged, so awaiting driver release.
will update this issue once again once the new driver is available.

**Comment by sfc-gh-igarish** [comment]
fix is released in 2.0.25
